app/scripts/controllers/serviceInstance.js

```diff
@@ -128,33 +128,39 @@ angular.module('openshiftConsole')
       updateParameterData();
     };
 
-    var updateServiceClass = function() {
-      // If we've previously loaded the service class or a request is in flight, don't do anything.
-      if (!$scope.serviceInstance || $scope.serviceClass || serviceClassPromise) {
-        return;
-      }
+    var updateServicePlan = function() {
+      Catalog.getServicePlansForServiceClass($scope.serviceClass).then(function (plans) {
```

Comment:
This re-requests plans for the service class on every watch update, correct? It would be better to only request them once on page load and remember them. Watches can trigger often due to status updates on the instance when it's being provisioned.

Reply:
Updated.